LayoutKitTests/OverlayLayoutTests.swift

```diff
@@ -0,0 +1,239 @@
+                    } else if index > backgroundCount + 1 {
+                        // We have an overlay layout
+                        let overlaySize = CGSize(width: Measurement.overlayWidth, height: Measurement.overlayHeight)
+                        let frame = Constant.alignments[index - backgroundCount - 2].position(
+                            size: overlaySize,
+                            in: expectedRect)
+                        AssertEqualDensity(subLayout.frame, [1.0: frame, 2.0: frame, 3.0: frame])
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     Tests that make views creates the correct number of views when there's different amounts of
+     background and overlay layouts passed.
+     */
+    func testViewCreationAndConfiguration() {
+        (0...MaxLayout.background).forEach { backgroundCount in
+            (0...MaxLayout.overlay).forEach { overlayCount in
+                var configCount = 0
+                let layout = SizeLayout<View>(width: 1, height: 1, config: { _ in
+                    configCount += 1
+                })
+                let overlay = OverlayLayout(
+                    primary: layout,
+                    background: [Layout](repeating: layout, count: backgroundCount),
+                    overlay: [Layout](repeating: layout, count: overlayCount),
+                    config: { _ in
+                        configCount += 1
+                })
+                let overlayView = overlay.arrangement().makeViews()
+                XCTAssertNotNil(overlayView)
+
+                // overlay config + primary config + backgroundCount + overlayCount
+                XCTAssertEqual(configCount, 2 + backgroundCount + overlayCount)
+            }
+        }
+    }
+
+    // MARK: - helpers
+
+    /**
+     Given the passed-in parameters, creates a complex overlay layout.
+     */
+    private func createComplexLayout(backgroundCount: Int,
+                                     overlayCount: Int,
+                                     primaryAlignment: Alignment) -> Layout {
+        let backgroundLayouts = (0...backgroundCount).map { index in
+            return SizeLayout(
+                width: Measurement.backgroundWidth,
+                height: Measurement.backgroundHeight,
+                alignment: Constant.alignments[index])
+        }
+        let overlayLayouts = (0...overlayCount).map { index in
+            return SizeLayout(
+                width: Measurement.overlayWidth,
+                height: Measurement.overlayHeight,
+                alignment: Constant.alignments[index])
+        }
+        let baseLayout = SizeLayout(
+            width: Measurement.primaryWidth,
+            height: Measurement.primaryHeight)
+        return OverlayLayout(
+            primary: baseLayout,
+            background: backgroundLayouts,
+            overlay: overlayLayouts,
+            alignment: primaryAlignment)
+    }
+}
```

Comment:
for your second question, already have it in `testComplexLayoutWithAlignment`. The name `primaryAlignment` is confusing. It's actually `overlayAlignment`. I changed the name of the variable throughout the test file.